.github/workflows/agent-build-new.yml

```diff
@@ -62,3 +76,30 @@ jobs:
           status: ${{ job.status }}
           steps: ${{ toJson(steps) }}
           channel: '#eng-dataset-cloud-tech'
+
+  prepare-artifacts:
+    runs-on: ubuntu-20.04
+    needs: build-linux-packages
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: /tmp/all_artifacts
+
+
+      - name: Prepare artifacts
+        run: |
+          mkdir -p /tmp/result_artifacts
+          cp -a /tmp/all_artifacts/linux-packages-*/. /tmp/result_artifacts
+          cp -a /tmp/all_artifacts/tarball-noarch/. /tmp/result_artifacts
+
+      - name: Save result artifact
+        uses: actions/upload-artifact@v3
```

Comment:
This is new to me, neat!  I'm guessing that artifacts like this are only accessible via actions/download-artifact ?

One thing that might be handy (to be done later) is an brief README that describes the overall build process

Reply:
I added a comment to clarify what is going on here. In short, we just pack all build artifacts into one single artifact, so when doing a release, our release pipieline does not have to download all artifacts one by one and get all of them at once.